### PR TITLE
refactor(pageserver): remove aux v1 configs

### PR DESCRIPTION
**Pull request by skyzh**
## Problem

Part of https://github.com/neondatabase/neon/issues/8623

## Summary of changes

Removed all aux-v1 config processing code. Note that we persisted it into the index part file, so we cannot really remove the field from index part. I also kept the config item within the tenant config, but we will not read it any more.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/tenant/config.rs [hunk 1]
```diff
@@ -410,9 +405,7 @@ impl TenantConfOpt {
             image_layer_creation_check_threshold: self
                 .image_layer_creation_check_threshold
                 .unwrap_or(global_conf.image_layer_creation_check_threshold),
-            switch_aux_file_policy: self
-                .switch_aux_file_policy
-                .unwrap_or(global_conf.switch_aux_file_policy),
+            switch_aux_file_policy: None,
```
Comment (VladLazar) [inline pageserver/src/tenant/config.rs:408]: Why do we have to keep this? Tenant configs are persisted on disk, but could we just ignore the field when de-serializing?

File: control_plane/src/pageserver.rs [hunk 2]
```diff
@@ -399,16 +399,12 @@ impl PageServerNode {
...
-                .remove("switch_aux_file_policy")
-                .map(|x| x.parse::<AuxFilePolicy>())
-                .transpose()
-                .context("Failed to parse 'switch_aux_file_policy'")?,
             lsn_lease_length: settings.remove("lsn_lease_length").map(|x| x.to_string()),
             lsn_lease_length_for_ts: settings
                 .remove("lsn_lease_length_for_ts")
                 .map(|x| x.to_string()),
         };
+        settings.remove("switch_aux_file_policy");
```
Comment (VladLazar) [inline control_plane/src/pageserver.rs:407]: Why do we have to `remove` this? Tests should use this config anymore, right?

File: control_plane/src/pageserver.rs [hunk 3]
```diff
@@ -499,18 +495,15 @@ impl PageServerNode {
...
-                    .transpose()
-                    .context("Failed to parse 'switch_aux_file_policy'")?,
                 lsn_lease_length: settings.remove("lsn_lease_length").map(|x| x.to_string()),
                 lsn_lease_length_for_ts: settings
                     .remove("lsn_lease_length_for_ts")
                     .map(|x| x.to_string()),
             }
         };
 
+        settings.remove("switch_aux_file_policy");
```
Comment (VladLazar) [inline control_plane/src/pageserver.rs:505]: Same question as above and why doesn't this function call into `parse_config` instead?

File: test_runner/performance/test_logical_replication.py [hunk 4]
```diff
@@ -9,15 +9,15 @@
...
+from fixtures.neon_fixtures import logical_replication_sync
 
 if TYPE_CHECKING:
     from fixtures.benchmark_fixture import NeonBenchmarker
     from fixtures.neon_api import NeonApiEndpoint
     from fixtures.neon_fixtures import NeonEnv, PgBin
 
 
-@pytest.mark.parametrize("pageserver_aux_file_policy", [AuxFileStore.V2])
+@pytest.mark.parametrize("pageserver_aux_file_policy")
```
Comment (VladLazar) [inline test_runner/performance/test_logical_replication.py:20]: You can remove this, no? Same for other occurrences.

**Comment by skyzh** [comment]
removed all occurrences of aux policy except in index part and ready for review :)

**Review by VladLazar — APPROVED** [review]
lgtm

**Comment by skyzh** [comment]
will merge after next week, I'd like to make https://github.com/neondatabase/infra/pull/2127 go through first.

**Pull request merged**